curly/main.go

```diff
@@ -121,51 +121,56 @@ func logErr(err error) {
 var _ io.Writer = (*Chunked)(nil)
 
 type Chunked struct {
-	Prefix   string
-	F        *os.File
-	idx      int
-	fileSize int
+	w              io.WriteCloser
+	size           int
+	maxSize        int
+	floppyCreateFn func() (io.WriteCloser, error)
 }
 
-func NewChunked(p string) (*Chunked, error) {
-	i := 0
-	f, err := os.Create(fmt.Sprintf("%s.%d", p, i))
+func fileFloppyFn(p string) func() (io.WriteCloser, error) {
+	i := -1
+	prefix := p
+	return func() (io.WriteCloser, error) {
+		i++
+		return os.Create(fmt.Sprintf("%s.%d", prefix, i))
+	}
+}
+
+func NewChunked(prefix string) (*Chunked, error) {
+	floppyFunc := fileFloppyFn(prefix)
+	w, err := floppyFunc()
```

Comment:
nit: code will be nice if you also do first invocation of `floppyFunc` lazily in Write method